src/dataModels/JSON.js

```diff
@@ -363,45 +358,8 @@ var JSON = DataModel.extend('dataModels.JSON', {
         this.pipeline.length = newLength || 0;
     },
 
-    /**
-     * Add a pipe to the data source pipeline.
-     * @desc No-op if already added.
-     * @param {dataSourcePipelineObject} newPipe
-     * @param {string|null|undefined} [afterPipe] - One of:
-     * * `null` - Inserts at beginning.
-     * * *string* - Name of an existing pipe _after which_ the new pipe will be added.
-     * * *else* _(including `undefined` or omitted)_ - Adds to end.
-     * @memberOf dataModels.JSON.prototype
-     */
-    addPipe: function(newPipe, referencePipe) {
-        var referenceIndex,
-            added = this.pipeline.find(function(pipe) { return pipe.type === newPipe.type; });
-
-        if (!added) {
-            if (referencePipe === null) {
-                referenceIndex = 0; // add to beginning
-            } else if (
-                !this.pipeline.find(function(pipe, index) {
-                    referenceIndex = index + 1; // add after found pipe
-                    return pipe.type === referencePipe;
-                })
-            ) {
-                referenceIndex = this.pipeline.length; // not found: add to end
-            }
-            this.pipeline.splice(referenceIndex, 0, newPipe);
-        }
-    },
-
     isDrillDown: function(event) {
-        return this.pipeline.find(function(pipe) {
-            var test = pipe.test,
-                type = typeof test;
-
-            test = type === 'function' && pipe.test ||
-                type === 'string' && this[pipe.test];
-
-            return test && test.call(this, event);
-        }.bind(this));
+        return this.dataSource.isDrillDown(event);
```

Comment:
replace `event` with `event.gridData.x` per my comment in https://github.com/openfin/hyper-analytics/pull/29#discussion-diff-78241725R275